train.py

```diff
@@ -29,6 +29,11 @@
 )
 from torchtitan.profiling import maybe_enable_memory_snapshot, maybe_enable_profiling
 
+try:
+    from torch.distributed.utils import _sync_module_states_with_mesh
+except ImportError:
+    pass
```

Comment:
I guess this will error downstream if trying to use HSDP.

Reply:
ye, I haven't done a very good protection yet. Add to this PR's todo.

Reply:
if you plan to land with `except ImportError`, i recommend at least logging a warning that tells users what they need to do (get pytorch newer than x?) to fix it

Reply:
Probably no, it looks like we are going to go with setting the seed. So no need to do the import.